lib/twitch/client.rb

```diff
@@ -139,5 +128,29 @@ def initialize_response(data_class, http_response)
         http_response
       end
     end
+
+    def renew_authorization_header
+      CONNECTION.headers['Authorization'] = "Bearer #{access_token}"
+    end
+
+    def request(http_method, *args)
+      Retriable.with_context(:twitch) do
+        CONNECTION.public_send http_method, *args
+      end
+    end
+
+    def require_access_token
```

Comment:
I understand that this method provides convenience for access tokens that are expired, but is this something that could be specified as an opt-in/opt-out parameter for initializing the client?

Reply:
Sorry, I don't understand you completely.

I've added spec for authentication by `#get_bits_leaderboard` method. Do you disagree with some of them?

Reply:
From looking at your library for auth you hooked in, the `@oauth2_client.refreshed_tokens` would add an additional request to attempt to gain a new access token. This may not always be the desired behavior, so I think it should be either optional or at least have the ability to toggle off, which would then return the expected 401 error.

Reply:
@mauricew I've rejected tokens refresh for `token_type: :application` (it's a default) and added specs for it, what do you think?

I mean, if user specify `token_type: :user` for concrete needed API end-points — I think they'd rather to see login request instead of just error.

Reply:
@AlexWayfer I'm also looking at this from the context of a web server, where the use of having an output to the console and getting the code from $stdin will not be an ideal way of handling the error. If anything, the link can be added as some kind of metadata to the response or added to a separately thrown ApiError.

Reply:
> @AlexWayfer I'm also looking at this from the context of a web server, where the use of having an output to the console and getting the code from $stdin will not be an ideal way of handling the error. If anything, the link can be added as some kind of metadata to the response or added to a separately thrown ApiError.

To be honest, I've taken the approach from this: https://developers.google.com/youtube/v3/quickstart/ruby

But I can agree: meta-data and custom outer CLI output would be better.

Unfortunately, I can't make a "real" PR as "draft" on GitHub, but I'll try to remake this.

Reply:
@mauricew now it's done. What do you think?

Reply:
Scanned over the changes and it's good so far, I'll have a further look done by Friday.